src/main/java/io/vertx/core/spi/cluster/DeliveryStrategy.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2011-2019 Contributors to the Eclipse Foundation
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+ * which is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ */
+
+package io.vertx.core.spi.cluster;
+
+import io.vertx.core.Promise;
+import io.vertx.core.Vertx;
+import io.vertx.core.eventbus.Message;
+
+/**
+ * {@link io.vertx.core.eventbus.EventBus Clustered EventBus (CEB)} delivery strategy.
+ * <p>
+ * This strategy is skipped by the <em>CEB</em> only when the user raises the {@link io.vertx.core.eventbus.DeliveryOptions#setLocalOnly(boolean)} flag.
+ * Consequently, implementations must be aware of local {@link io.vertx.core.eventbus.EventBus} registrations.
+ */
+public interface DeliveryStrategy {
```

Comment:
Overall I'm not comfortable with the naming `DeliveryStrategy` that is very generic. I can see that the main responsibility of this interface is to choose a node and in fact the default implementation delegates to the `Selectors` class.

I believe we should simply rename this interface `NodeSelector`.

Reply:
The default implementation in vertx-core is intentionally "simple": it reproduces the behaviour that exists in Vert.x 3.

However the goal of this extension point it to allow users with specific needs to implement advanced scenarios:

- sending to nodes in the same rack
- avoiding nodes which are overloaded
- partitioning (based on message headers)

The name strategy sounded appropriate to me to describe this but if that's a blocker I can change it.

Reply:
I am not keen on the precise `NodeSelector` name, it was a proposal (inspired from the `Selectors` class). Delivery strategy is not a great fit here: this SPI allows to have custom implementations of choosing a list of node(s) provided by a cluster manager to send a given message to.

Reply:
To me it really sounds like a strategy: ours (the default) is round robin, users could implement any other they see fit.

Reply:
I've renamed it to `NodeSelector`.